.github/workflows/test_tap.yml

```diff
@@ -15,7 +15,7 @@ jobs:
       GITHUB_TOKEN: ${{secrets.GITHUB_TOKEN}}
     strategy:
       matrix:
-        python-version: [3.7, 3.8, 3.9, "3.10"]
+        python-version: ["3.10", 3.9, 3.8, 3.7]
```

Comment:
I realised the lower versions are more likely to crash, so I reordered them to run from least likely to most likely to crash (so you can recognise if your code works on later versions earlier :eyes:)

Reply:
yup up to you if you want to disable it. Not too sure how github credits work for this project but if it's plentiful, I reckon it's a good idea to disable it.

Reply:
The only question is - does the caching work between python versions? I am not sure how the matrix handles caching tbh. If it does then we can remove the fail-fast option. Otherwise we should keep it